### ISSUE TITLE
Example Dockerfile for static build

### ISSUE THREAD

**Issue by cedricve** [hybridgroup/gocv#1050]
As suggested in a few PR's and `Makefile`, it would be beneficial to add a working static build `Dockerfile`. I have been trying to create the static OpenCV build (which works). But when using the `static` tag while running `go build`, I'm getting a few linker errors. I'm currently doing this on an ARM64 (Mac M1), which might explain the linker issues to IPP. I wonder how we could make this work for othe architectures like ARMv7,ARM64,etc. 

The static base image:

      FROM golang:1.19-buster 
      LABEL maintainer="hybridgroup"

      RUN apt-get update && apt-get install -y --no-install-recommends \
            git build-essential cmake pkg-config unzip libgtk2.0-dev \
            curl ca-certificates libcurl4-openssl-dev libssl-dev \
            libavcodec-dev libavformat-dev libswscale-dev libtbb2 libtbb-dev \
            libjpeg62-turbo-dev libpng-dev libtiff-dev libdc1394-22-dev && \
            rm -rf /var/lib/apt/lists/*

      ARG OPENCV_VERSION="4.7.0"
      ENV OPENCV_VERSION $OPENCV_VERSION

      ARG OPENCV_FILE="https://github.com/opencv/opencv/archive/${OPENCV_VERSION}.zip"
      ENV OPENCV_FILE $OPENCV_FILE

      ARG OPENCV_CONTRIB_FILE="https://github.com/opencv/opencv_contrib/archive/${OPENCV_VERSION}.zip"
      ENV OPENCV_CONTRIB_FILE $OPENCV_CONTRIB_FILE

      RUN curl -Lo opencv.zip ${OPENCV_FILE} && \
            unzip -q opencv.zip && \
            curl -Lo opencv_contrib.zip ${OPENCV_CONTRIB_FILE} && \
            unzip -q opencv_contrib.zip && \
            rm opencv.zip opencv_contrib.zip

      RUN cd opencv-${OPENCV_VERSION} && \
            mkdir build && cd build && \
            cmake -D CMAKE_BUILD_TYPE=RELEASE \
            -D CMAKE_INSTALL_PREFIX=/usr/local \
            -D OPENCV_EXTRA_MODULES_PATH=../../opencv_contrib-${OPENCV_VERSION}/modules \
            -D BUILD_SHARED_LIBS=OFF \
            -D ENABLE_NEON=ON \
            -D WITH_FFMPEG=ON \
            -D WITH_TBB=ON \
            -D BUILD_TBB=ON \
            -D BUILD_TESTS=OFF \
            -D WITH_EIGEN=OFF \
            -D WITH_GSTREAMER=OFF \
            -D WITH_V4L=ON \
            -D WITH_LIBV4L=ON \
            -D WITH_VTK=OFF \
            -D WITH_QT=OFF \
            -D BUILD_JPEG=ON \
            -D OPENCV_ENABLE_NONFREE=ON \
            -D BUILD_DOCS=OFF \
            -D BUILD_EXAMPLES=OFF \
            -D BUILD_TESTS=OFF \
            -D BUILD_PERF_TESTS=OFF \
            -D BUILD_opencv_java=NO \
            -D BUILD_opencv_python=NO \
            -D BUILD_opencv_python2=NO \
            -D BUILD_opencv_python3=NO \
            -D OPENCV_GENERATE_PKGCONFIG=ON .. && \
            make -j $(nproc --all) && \
            make preinstall && make install && ldconfig && \
            cd / && rm -rf opencv*

The static build container

	# to build this docker image:
	#   docker build .
	FROM gocv- base

	ENV GOPATH /go

	COPY . /go/src/gocv.io/x/gocv/

	WORKDIR /go/src/gocv.io/x/gocv
	RUN go build -tags static --ldflags '-extldflags "-static"' -o static_version ./cmd/version/main.go

	CMD ["/build/gocv_version"]

While creating the build

    > [4/4] RUN go build -tags static --ldflags '-extldflags "-static"' -o static_version ./cmd/version/main.go:                                                                                                   
    #6 30.26 # command-line-arguments                                                                       
    #6 30.26 /usr/local/go/pkg/tool/linux_arm64/link: running g++ failed: exit status 1                     
    #6 30.26 /usr/bin/ld: cannot find -lippiw                                                               
    #6 30.26 /usr/bin/ld: cannot find -lippicv
    #6 30.26 /usr/bin/ld: /usr/local/lib/libopencv_core.a(opencl_core.cpp.o): in function `opencl_check_fn(int)':
    #6 30.26 opencl_core.cpp:(.text._ZL15opencl_check_fni+0x1b0): warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
    #6 30.26 /usr/bin/ld: cannot find -lquadmath
    #6 30.26 collect2: error: ld returned 1 exit status
    #6 30.26

**Comment by cedricve** [comment]
Update
----
 I made a static cross build for OpenCV with `docker buildx` (https://hub.docker.com/r/kerberos/gocv/tags), and I confirm that buiding using `--platform linux/amd64` does not work properly as well, so there would need to come a fix for all architectures (including AMD64). Anyone an idea? We would like this to use static builds of our open source [Kerberos Agent](https://github.com/kerberos-io/agent), without the need for installing Docker, or if not possible to utilise Docker.




	#0 266.5 parallel_tbb.cpp:(.text._ZNK2cv8parallel3tbb18ParallelForBackend13getNumThreadsEv[_ZNK2cv8parallel3tbb18ParallelForBackend13getNumThreadsEv]+0x50): undefined reference to `tbb::interface7::internal::task_arena_base::internal_max_concurrency(tbb::interface7::task_arena const*)'
	#0 266.5 /usr/bin/ld: parallel_tbb.cpp:(.text._ZNK2cv8parallel3tbb18ParallelForBackend13getNumThreadsEv[_ZNK2cv8parallel3tbb18ParallelForBackend13getNumThreadsEv]+0xbc): undefined reference to `tbb::interface7::internal::task_arena_base::internal_max_concurrency(tbb::interface7::task_arena const*)'
	#0 266.5 /usr/bin/ld: /usr/local/lib/libopencv_core.a(parallel_tbb.cpp.o): in function `cv::parallel::tbb::ParallelForBackend::setNumThreads(int)':
	#0 266.5 parallel_tbb.cpp:(.text._ZN2cv8parallel3tbb18ParallelForBackend13setNumThreadsEi[_ZN2cv8parallel3tbb18ParallelForBackend13setNumThreadsEi]+0xc0): undefined reference to `tbb::interface7::internal::task_arena_base::internal_terminate()'
	#0 266.5 /usr/bin/ld: parallel_tbb.cpp:(.text._ZN2cv8parallel3tbb18ParallelForBackend13setNumThreadsEi[_ZN2cv8parallel3tbb18ParallelForBackend13setNumThreadsEi]+0xee): undefined reference to `tbb::interface7::internal::task_arena_base::internal_initialize()'
	#0 266.5 /usr/bin/ld: /usr/local/lib/libopencv_core.a(parallel_tbb.cpp.o):(.data.rel.ro._ZTIN3tbb10interface98internal9start_forINS_13blocked_rangeIiEEN2cv8parallel3tbb18ParallelForBackend13CallbackProxyEKNS_16auto_partitionerEEE[_ZTIN3tbb10interface98internal9start_forINS_13blocked_rangeIiEEN2cv8parallel3tbb18ParallelForBackend13CallbackProxyEKNS_16auto_partitionerEEE]+0x10): undefined reference to `typeinfo for tbb::task'
	#0 266.5 /usr/bin/ld: $WORK/b001/exe/a.out: hidden symbol `opj_stream_default_create' isn't defined

**Comment by J7mbo** [comment]
Did you ever get this working?

**Comment by cedricve** [comment]
Yes we made it working for the [kerberos.io project](https://github.com/kerberos-io) in this base image https://github.com/kerberos-io/base/blob/main/Dockerfile-opencv-armv7.

However we got rid of it in the current implementation as we were just using a fraction of the functionalities (motion detection) in our [Kerberos Agent](https://github.com/kerberos-io/agent); a video surveillance solution.
